packages/commerce-sdk-react/src/hooks/ShopperOrders/mutation.ts

```diff
@@ -81,19 +80,43 @@ export const SHOPPER_ORDERS_NOT_IMPLEMENTED = [
     'UpdatePaymentInstrumentForOrder'
 ]
 
+export type ShopperOrdersMutationType = typeof ShopperOrdersMutations[keyof typeof ShopperOrdersMutations]
+
+type UseShopperOrdersMutationHeaders = NonNullable<Argument<Client['createOrder']>>['headers']
+type UseShopperOrdersMutationArg = {
+    headers?: UseShopperOrdersMutationHeaders
+    rawResponse?: boolean
+    action: ShopperOrdersMutationType
+}
+
 /**
  * A hook for performing mutations with the Shopper Orders API.
  */
-export function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(action: Action) {
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg & {rawResponse?: false}
+): UseMutationResult<DataType<Client[Action]>, Error>
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg & {rawResponse: true}
+): UseMutationResult<Response, Error>
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg
+): UseMutationResult<DataType<Client[Action]>, Error, Argument<Client[Action]>> {
+    const {headers, rawResponse, action} = arg
+
     if (SHOPPER_ORDERS_NOT_IMPLEMENTED.includes(action)) {
         NotImplemented()
     }
     type Params = Argument<Client[Action]>
     type Data = DataType<Client[Action]>
     const queryClient = useQueryClient()
+
     return useMutation<Data, Error, Params>(
         (params, apiClients) => {
             const method = apiClients['shopperOrders'][action] as MutationFunction<Data, Params>
+            //Set headers to user-defined header values
+            if (params) {
+                params.headers = headers
+            }
             return method.call(apiClients['shopperOrders'], params)
```

Comment:
I expected the `rawResponse` to be used like this: `method.call(apiClients['shopperOrders'], params, rawResponse)`

But Typescript is complaining about the extra argument 🤔 That's strange... it should be correct according to the documentation. 

![Google Chrome 2022-12-06 at 17 14 14](https://user-images.githubusercontent.com/847300/206063552-8756ee15-bdd0-4bb5-8bec-0b6120142d75.png)


Maybe on line 115 above, we're not assigning the correct type?

Reply:
I also was thinking to pass in `rawResponse` like so:
`method.call(apiClients['shopperOrders'], params, rawResponse)`

but am facing challenges with the TS types and restrictions. When I pass in `rawResponse` an error occurs that says there are only 2 arguments allowed. Not sure where this is being set.